src/content/layout/builtin_layout.cc

```diff
@@ -51,6 +51,74 @@
 
 #endif //ONLINE_SERVICES
 
+BuiltinLayout::BuiltinLayout(std::shared_ptr<ContentManager> content)
+    : Layout(std::move(content))
+    , config(this->content->getContext()->getConfig())
+    , genreMap(this->config->getDictionaryOption(CFG_IMPORT_SCRIPTING_IMPORT_GENRE_MAP))
+{
+    container["Video"] = std::make_shared<CdsContainer>("Video");
+    container["All Video"] = std::make_shared<CdsContainer>("All Video");
+    chain["/Video/All Video"] = this->content->addContainerTree(std::vector<std::shared_ptr<CdsObject>> { container["Video"], container["All Video"] });
```

Comment:
the vector param can be removed. {} is enough.

I am curious what happens here though. std::initializer_list is a view into an array. not an array itself. So these shared pointers get created and then copied.

I've hit this issue before in this codebase where std::vector {std::make_shared x} warned about a memory leak.

Maybe the proper solution is to make the function parameter an initializer_list. I don't know the ins and outs of initializer_list that well.

Reply:
Yeah...

https://godbolt.org/z/7x4Tjc4fP

The commented part does not compile.

Reply:
Your example uses `unique_ptr`. On the other hand, I'm not creating the `shared_ptr` in line, it's already in the container map.

Reply:
It uses unique_ptr to see when copies happen. Well anyway, maybe fanalyzer or clang-analyzer can detect if there's an issue.